qip-0015.md

```diff
@@ -0,0 +1,59 @@
+```
+  QIP: 15
+  Layer: Consensus (hard fork) 
+  Title: Qi UTXO Set Trimming
+  Author: jdowning100
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/quai-network/qips/wiki/Comments:QIP-00015
+  Status: Active
+  Type: Standards Track
+  Created: 2024-09-06
+  License: BSD-2-Clause
+```
+
+## Abstract
+
+The UTXO set is the set of all unspent transaction outputs in the Qi ledger. Qi is a money system with cash-like properties, such as denominations that can be broken down into smaller denominations. Combining smaller denominations into larger ones can be done with a different mechanism, but that is to be discussed in a separate QIP. This QIP proposes a mechanism to regularly trim dust UTXOs from the UTXO set.
+
+## Motivation
+
+The Qi UTXO set must be held on the disk of every node in the network in order for nodes to come into consensus, and the set can grow infinitely large. There are currently about [187.3 million UTXOs](https://www.blockchain.com/explorer/charts/utxo-count) in Bitcoin, with at ~70ish bytes per UTXO, takes up about 13gb of space on disk. 
+I fully believe that to build a monetary system for every individual the UTXO set for a given zone could grow to as large as 20 billion, which is about 200 UTXOs per user for 100 million users per zone. Qi UTXOs are about 57 bytes: 2 bytes for the leveldb prefix, 32 bytes for the txhash, 2 bytes for the index, 20 bytes for the address that may spend the UTXO, 1 byte for the denomination, and 1 byte or more for the lock (if applicable). This means that the UTXO set on disk could grow to as large as 1.14 TB. 
```

Comment:
Need some math to back up your assumptions here. e.g. How do you determine every user needs 200 UTXOs? How did you arrive at 100M users per zone?

Reply:
I don't think it's worthwhile in this QIP to explain why a user needs a certain amount of UTXOs or how many users we should support. That debate should happen elsewhere. These numbers are reasonable. If you don't think they are reasonable, you may explain why, but again, it should be done in a forum post or something.